### PR TITLE
✨ Added extension code for amp-mowplayer [NEW]

### PR DESCRIPTION
**Pull request by gopanisandip**
I have added my amp-mowplayer code in this new branch because with previous PR we got some issues in merging code.

**Intent to Implement:**
https://github.com/ampproject/amphtml/issues/16696

**Old PR :**
https://github.com/ampproject/amphtml/pull/16697

@aghassemi

### CODE REVIEW

File: extensions/amp-mowplayer/validator-amp-mowplayer.protoascii [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the license.
+#
+
+tags: {  # amp-mowplayer
+  html_format: AMP
+  tag_name: "SCRIPT"
+  satisfies: "amp-mowplayer extension .js script"
+  requires: "amp-mowplayer"
```
Comment (Gregable) [inline extensions/amp-mowplayer/validator-amp-mowplayer.protoascii]: `satisfies`/`requires` pairs are no longer required for basic extension / tag pairs. There may be some out of date documentation somewhere, but please remove.

File: extensions/amp-mowplayer/validator-amp-mowplayer.protoascii [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    allowed_versions: "0.1"
+    allowed_versions: "latest"
+  }
+  attr_lists: "common-extension-attrs"
+}
+tags: {  # <amp-mowplayer>
+  html_format: AMP
+  tag_name: "AMP-MOWPLAYER"
+  satisfies: "amp-mowplayer"
+  requires: "amp-mowplayer extension .js script"
```
Comment (Gregable) [inline extensions/amp-mowplayer/validator-amp-mowplayer.protoascii]: same here, please remove

File: extensions/amp-mowplayer/validator-amp-mowplayer.protoascii (hunk 1 above)
Comment (gopanisandip) [reply extensions/amp-mowplayer/validator-amp-mowplayer.protoascii]: Sure, Thanks!

File: extensions/amp-mowplayer/validator-amp-mowplayer.protoascii (hunk 2 above)
Comment (gopanisandip) [reply extensions/amp-mowplayer/validator-amp-mowplayer.protoascii]: Sure, Thanks!

**Comment by aghassemi** [comment]
Thanks @gopanisandip there are a few lint errors to fix, see https://travis-ci.org/ampproject/amphtml/jobs/412965039.

**Comment by aghassemi** [comment]
@alanorozco This is same PR as https://github.com/ampproject/amphtml/pull/16697 which had rebasing issue.

**Comment by Gregable** [comment]
Validator changes look good.

**Comment by gopanisandip** [comment]
@aghassemi  Fixed link error shown in https://travis-ci.org/ampproject/amphtml/jobs/412965039

File: extensions/amp-mowplayer/0.1/amp-mowplayer.js [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+
+  /**
+   * @return {string}
+   * @private
+   */
+  getVideoIframeSrc_() {
+    if (this.videoIframeSrc_) {
+      return this.videoIframeSrc_;
+    }
+    const params = dict({
```
Comment (aghassemi) [inline extensions/amp-mowplayer/0.1/amp-mowplayer.js]: this needs to be casted using `/** @type {!JsonObject} */ dict(.....`

File: extensions/amp-mowplayer/0.1/amp-mowplayer.js [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+
+  /**
+   * @return {string}
+   * @private
+   */
+  getVideoIframeSrc_() {
+    if (this.videoIframeSrc_) {
+      return this.videoIframeSrc_;
+    }
+    const params = /** @type {!JsonObject} */ (dict({
```
Comment (alanorozco) [inline extensions/amp-mowplayer/0.1/amp-mowplayer.js]: No need to typecast here.

File: extensions/amp-mowplayer/0.1/amp-mowplayer.js [hunk 5]
```diff
@@ -0,0 +1,367 @@
...
+      'code': this.mediaid_,
+    }));
+    const src = addParamsToUrl('https://cdn.mowplayer.com/player.html',params);
```
Comment (alanorozco) [inline extensions/amp-mowplayer/0.1/amp-mowplayer.js]: comma before `params`

Comment (gopanisandip) [reply extensions/amp-mowplayer/0.1/amp-mowplayer.js]: So basically need to remove typecast from line 129 and need to add typecast to line 132 like :

**Comment by aghassemi** [comment]
@gopanisandip Please add your GitHub username to `extensions/amp-mowplayer/OWNERS.yaml` and refresh the PR from master ( `git pull upstream master; git push origin <nameof yourbrach>` ) and we should be good to go.

@alanorozco @Gregable PTAL.

**Comment by Gregable** [comment]
LGTM for validator changes.

**Comment by aghassemi** [comment]
Thanks @Gregable, could you please approve? (still blocked on `requested changes`)

**Comment by gopanisandip** [comment]
@alanorozco Could you please give me detail about next step? when this will be include with live amp? what are the next step?

**Comment by alanorozco** [comment]
Please fix the validator output error [as failing on Travis](https://travis-ci.org/ampproject/amphtml/jobs/420208615). After you've addressed the issue, please ping me here so I can merge.

**Comment by Gregable** [comment]
In particular, this response in the travis output is probably all you need:
```If validator/../extensions/amp-mowplayer/0.1/test/validator-amp-mowplayer.out is incorrect, please run `gulp validator --update_tests` to regenerate it based on its corresponding .html file.```

**Comment by gopanisandip** [comment]
@alanorozco Could you please move forward with merging code? Let me know you need anything from my side.

**Comment by alanorozco** [comment]
Merged. Thanks for your contribution @gopanisandip!

**Pull request merged**